src/main/java/fi/okm/jod/yksilo/service/profiili/Mapper.java

```diff
@@ -124,7 +125,9 @@ public static YksilonOsaaminenDto mapYksilonOsaaminen(YksilonOsaaminen entity) {
   }
 
   private static OsaamisenLahdeDto mapOsaamisenLahde(@NonNull YksilonOsaaminen entity) {
-    return new OsaamisenLahdeDto(entity.getLahdeTyyppi(), entity.getLahde().getId());
+    return new OsaamisenLahdeDto(
+        entity.getLahdeTyyppi(),
+        Optional.ofNullable(entity.getLahde() != null ? entity.getLahde().getId() : null));
```

Comment:
if entity.getLahde() already returned an Optional, this would reduce to `entity.getLahde().map(OsaamisenLahde::getId)`